### PR TITLE
Add auto-merge app

### PR DESCRIPTION
**Pull request by bobvanderlinden**
Adds the probot-auto-merge app to the list.

The app allows automatically merging of PRs based on a number of conditions.

Conditions include labels, reviews and it checks test suites as well.

It also updates branches when a PR is based on a strict protected branch. It also deletes merged branches.

All of this is configurable using a file `.github/auto-merge.yml`.

See the project repository for more information: https://github.com/bobvanderlinden/probot-auto-merge

I have hosted the bot on a free Heroku instance. I have little experience with such usage and costs. Any suggestions on this front are welcome, but I'll just give it a try and move to something else or paid when needed.

##### Checklist

- [x] If you're adding a listing for your app, be sure your app is in the `/_apps/` folder before opening this Pull Request.
- [x] All comments in frontmatter need to be removed.
- [x] Performs a useful action through the GitHub API that solves an existing problem for developers: automatically merges PRs based on certain conditions (review approvals, labels, etc)
- [x] Is original: for example, it does something not already done by an existing Probot app. Have not found a Probot app doing the same thing.
- [x] [Has tests](https://github.com/bobvanderlinden/probot-auto-merge/tree/master/test) with [coverage](https://coveralls.io/github/bobvanderlinden/probot-auto-merge)
- [x] [Has documentation](https://github.com/bobvanderlinden/probot-auto-merge/blob/master/README.md)
- [x] [Is open source](https://github.com/bobvanderlinden/probot-auto-merge/)
- [x] [Has a license](https://github.com/bobvanderlinden/probot-auto-merge/blob/master/LICENSE)
- [x] [Has a code of conduct](https://github.com/bobvanderlinden/probot-auto-merge/blob/master/CODE_OF_CONDUCT.md)
- [x] Has someone willing to at least minimally maintain them for the near future: @bobvanderlinden

### CODE REVIEW

File: _apps/auto-merge.md [hunk 1]
```diff
@@ -0,0 +1,198 @@
...
+Configuration of probot-auto-merge is done through `.github/auto-merge.yml` in
+your repository. An example of this file can be found [here](https://github.com/bobvanderlinden/probot-auto-merge/auto-merge.example.yml).
+
+The configuration has values that serve as conditions on whether or not a pull request
+should be automatically merged and also configuration about the merge itself. Values
+that serve as conditions are annotated as such below.
+
+All conditions must be met before a PR will be automatically merged. You can get more
+flexibility by defining multiple rules. Rules can have multiple conditions and if any
+of the conditions inside a rule are met, the PR is also merged. See [rules](#Rules).
```
Comment (hiimbex) [inline _apps/auto-merge.md]: I think `[rules](#Rules)` needs to be `[rules](#`rules` (default: none))` to work

File: _apps/auto-merge.md [hunk 2]
```diff
@@ -0,0 +1,198 @@
+---
+title: Auto Merge
+description: Automatically merges pull requests based on conditions
```
Comment (hiimbex) [inline _apps/auto-merge.md]: `Automatically` is a banned word from descriptions, so maybe try:
> merges pull requests when configured conditions are met

**Comment by bobvanderlinden** [comment]
I've removed most of the documentation from the description / page, because I realised it'll become out-of-date rather quickly. It now contains a link to the right section in the readme of probot-auto-merge.

**Comment by bobvanderlinden** [comment]
ping @hiimbex

**Comment by bobvanderlinden** [comment]
Ah, alright. I initially thought `auto-merge` would be a confusing name without the `probot` prefix. I'll give this another look. Thanks for the feedback!

**Comment by hiimbex** [comment]
Well, you have to own the app posting you list, and we didn't want folks thinking certain apps were being explicitly supported or maintained by probot.

Names are hard though :(

**Comment by SvanBoxel** [comment]
Hey @bobvanderlinden, what's the status of your app? I've postponed the work on my own (stripped down) solution that does this when I noticed this PR. 👍

**Pull request closed without merging**